amqp/transport.py

```diff
@@ -360,29 +360,25 @@ def _wrap_socket_sni(self, sock, keyfile=None, certfile=None,
 
         opts = {
             'sock': sock,
-            'keyfile': keyfile,
-            'certfile': certfile,
             'server_side': server_side,
-            'cert_reqs': cert_reqs,
-            'ca_certs': ca_certs,
             'do_handshake_on_connect': do_handshake_on_connect,
             'suppress_ragged_eofs': suppress_ragged_eofs,
-            'ciphers': ciphers,
-            'ssl_version': ssl_version
+            'server_hostname': server_hostname,
         }
 
-        sock = ssl.wrap_socket(**opts)
+        context = ssl.SSLContext(ssl_version)
+        if certfile is not None:
+            context.load_cert_chain(certfile, keyfile)
+        if cert_reqs != ssl.CERT_NONE:
+            context.check_hostname = True
+        if (certfile is not None) and (keyfile is not None):
+            context.load_cert_chain(certfile, keyfile)
```

Comment:
This lines are redundant with the previous call to load_cert_chain a few lines above.

Reply:
Ah right good catch